src/convenience/session.ts

```diff
@@ -85,6 +85,30 @@ export interface StorageAdapter<T> {
      * Deletes a value for the given key from the storage.
      */
     delete: (key: string) => MaybePromise<void>;
+    /**
+     * Checks whether a key exists in the storage
+     */
+    has?: (key: string) => MaybePromise<boolean>;
+    /**
+     * Lists all keys
+     */
+    readAllKeys?: () => Iterable<string> | AsyncIterable<string>;
+    /**
+     * Lists all values
+     */
+    readAllValues?: () => Iterable<T> | AsyncIterable<T>;
+    /**
+     * Lists all keys with their values
+     */
+    readAllEntries?: () =>
+        | Iterable<[key: string, value: T]>
+        | AsyncIterable<[key: string, value: T]>;
+    /**
+     * Lists the values for all the keys that match the RegExp pattern
+     * @param pattern RegExp to match keys against
+     * @returns Values with a matching key
+     */
+    readManyValues?: (pattern: RegExp) => Iterable<T> | AsyncIterable<T>;
```

Comment:
I*m concerned about this one. Not only that there's no immediate use case for it, but mainly that executing regular expressions on potentially distributed data stores is a hard problem to solve. It seems like the most efficient solution for this would be to enumerate all keys, and then filter them down. However, if the data storage has no special support for it, then we should not implement this logic in every store, but rather do it plugin-side.

Can you elaborate why you added this, or revert if it's not necessary to have it?

Reply:
I'll do it on the plugin's side